cobalt/user_agent_platform_info.cc

```diff
@@ -0,0 +1,602 @@
+
+  // TODO(cobalt, b/374213479): should we get rasterizer_type from
+  // CobaltExtensionConfigurationApi?
+  //   std::string rasterizer_type_setting =
+  //       info.enable_skia_rasterizer()
+
+  //   std::string rasterizer_type =
+  //       renderer::GetDefaultRasterizerForPlatform(rasterizer_type_setting)
+  //           .rasterizer_name;
+  std::string rasterizer_type = "gles";
+  info.set_rasterizer_type(rasterizer_type);
+
+// Evergreen info
+#if SB_IS(EVERGREEN)
+  updater::EvergreenLibraryMetadata evergreen_library_metadata =
+      updater::GetCurrentEvergreenLibraryMetadata();
+  info.set_evergreen_version(evergreen_library_metadata.version);
+  info.set_evergreen_file_type(evergreen_library_metadata.file_type);
+  if (!SbSystemGetExtension(kCobaltExtensionInstallationManagerName)) {
+    // If the installation manager is not initialized, the "evergreen_lite"
+    // command line parameter is specified and the system image is loaded.
+    info.set_evergreen_type("Lite");
+  } else {
+    info.set_evergreen_type("Full");
+  }
+#endif
+
+  // Additional Platform Info
+  auto platform_info_extension =
+      static_cast<const CobaltExtensionPlatformInfoApi*>(
+          SbSystemGetExtension(kCobaltExtensionPlatformInfoName));
+  if (platform_info_extension) {
+    if (platform_info_extension->version >= 1) {
+      result = platform_info_extension->GetFirmwareVersionDetails(
+          value, starboard::kSystemPropertyMaxLength);
+      if (result) {
+        info.set_android_build_fingerprint(value);
+      }
+      info.set_android_os_experience(
+          platform_info_extension->GetOsExperience());
+    }
+    if (platform_info_extension->version >= 2) {
+      int64_t ver = platform_info_extension->GetCoreServicesVersion();
+      if (ver != 0) {
+        std::string sver = std::to_string(ver);
+        info.set_android_play_services_version(sver);
+      }
+    }
+  }
+
+  // TODO(cobalt, b/374213479): are we going to retrieve build_id and
+  // cobalt_version from cobalt/build/build_info.py?
+  std::string cobalt_version = "26.lts.0";
+  std::string build_id = "1234567";
+  info.set_cobalt_version(cobalt_version);
+  info.set_cobalt_build_version_number(build_id);
+
+  // TODO(cobalt, b/374213479): how do we retrieve build type macro?
+  // #if defined(COBALT_BUILD_TYPE_DEBUG)
+  //   info.set_build_configuration("debug");
+  // #elif defined(COBALT_BUILD_TYPE_DEVEL)
+  //   info.set_build_configuration("devel");
+  // #elif defined(COBALT_BUILD_TYPE_QA)
+  //   info.set_build_configuration("qa");
+  // #elif defined(COBALT_BUILD_TYPE_GOLD)
+  //   info.set_build_configuration("gold");
+  // #else
+  // #error Unknown build configuration.
+  // #endif
+  info.set_build_configuration("qa");
+
+  result = SbSystemGetProperty(kSbSystemPropertyUserAgentAuxField, value,
+                               starboard::kSystemPropertyMaxLength);
+  if (result) {
+    info.set_aux_field(value);
+  }
+
+#if SB_API_VERSION >= 15
```

Comment:
IIUC we don't need to support old versions of Starboard  
so you should be able to remove these provisions.